bundles/io/org.eclipse.smarthome.io.transport.mqtt/src/main/java/org/eclipse/smarthome/io/transport/mqtt/MqttService.java

```diff
@@ -222,17 +231,17 @@ public boolean addBrokerConnection(MqttBrokerConnection connection) {
      * @throws ConfigurationException Most likely your provided name and url are invalid.
      * @throws MqttException
      */
-    public MqttBrokerConnection addBrokerConnection(Map<String, String> brokerConnectionConfig)
+    public @Nullable MqttBrokerConnection addBrokerConnection(Map<String, String> brokerConnectionConfig)
             throws ConfigurationException, MqttException {
         // Extract mandatory fields
         String brokerID = brokerConnectionConfig.get(NAME_PROPERTY);
-        if (StringUtils.isBlank(brokerID)) {
+        if (brokerID == null || brokerID.isEmpty()) {
```

Comment:
Imho it really is a pity that we have to "work around" the annotations now just because the compiler is not able to recognise `StringUtils.isBlank` as an implicit null check. I just had a similar situation in a builder pattern (see https://github.com/eclipse/smarthome/pull/4301/files#diff-8f975ae88094089585192875908ba08bR79) which I could solve by doing some API changes. In the end it turned out these changes where a turn for the better.

Just to clarify: I thought `@NonNullByDefault` should guarantee the call to `brokerConnectionConfig.get(NAME_PROPERTY)` to always return a non null value? So why is the null check (or StringUtils.isBlank) required at all?

Reply:
> Imho it really is a pity that we have to "work around" the annotations now just because the compiler is not able to recognise `StringUtils.isBlank` as an implicit null check.

You are right, the Eclipse Nullness Checker should be improved and the EEAs.
The Checker Framework can handle such situations already.
There I could use astub files (similar to EEA) and state that if the method returns true, the reference used in the argument is non-null.

```Java
class StringUtils {
    static @EnsuresNonNull("#1") boolean isBlank(@Nullable String str);
}
```